ph5/utilities/ph5validate.py

```diff
@@ -749,47 +756,47 @@ def check_event_t_completeness(self, event):
+                    error.append("Event ID '%s' not between 0 and 65535."
+                                 % event['id_s'])
+                elif int(event['id_s']) > 32767:
+                    warning.append("Event ID '%s' is more than 32767. "
+                                   "Not compatible with SEGY revision 1."
+                                   % event['id_s'])
+            except ValueError:
+                error.append("Event ID '%s' not a whole "
+                             "number between 0 and 65535." % event['id_s'])
+
         if not event['description_s']:
             warning.append("Event description is missing.")
         # EVENT LOCATION
-        if event['location/coordinate_system_s'] is None:
-            warning.append("No Event location/coordinate_system_s value "
-                           "found.")
-
-        if event['location/projection_s'] is None:
-            warning.append("No Event location/projection_s value "
-                           "found.")
-
-        if event['location/ellipsoid_s'] is None:
-            warning.append("No Event location/ellipsoid_s value "
-                           "found.")
-
-        if event['location/description_s'] is None:
-            warning.append("No Event location/description_s value "
-                           "found.")
-
+        # removed check for coordinate_system_s, projection_s,
+        # projection_s , ellipsoid_s , description_s
+        # because they are not required and most of PIs do not fill them out
         if event['location/X/value_d'] == 0:
             error.append("Event location/X/value_d "
                          "'longitude' seems to be 0. "
                          "Is this correct???")
-        if event['location/X/units_s'] is None:
-            error.append("No Event location/X/units_s value "
-                         "found.")
+        if event['location/X/units_s'] in [None, '']:
+            warning.append("No Event location/X/units_s value "
+                           "found.")
 
         if event['location/Y/value_d'] == 0:
             error.append("Event location/Y/value_d "
                          "'latitude' seems to be 0. "
                          "Is this correct???")
-        if event['location/X/units_s'] is None:
-            error.append("No Event location/Y/units_s value "
-                         "found.")
+        if event['location/Y/units_s'] in [None, '']:
+            warning.append("No Event location/Y/units_s value "
+                           "found.")
 
         if not event['location/Z/value_d']:
             error.append("No Event location/Z/value_d value "
                          "found.")
-        if event['location/X/units_s'] is None:
-            error.append("No Event location/Z/units_s value "
-                         "found.")
+        if event['location/Z/units_s'] in [None, '']:
+            warning.append("No Event location/Z/units_s value "
+                           "found.")
```

Comment:
Does it make more sense to turn missing location information in the array table into an error rather than turning missing event location information into a warning?

Doesn't the event need location information? I would think an event's location would be important from a scientific prescriptive.

Reply:
When noven is used to turn an event csv into a kef, the X,Y,Z/unit_s are not available in the dropdown options, but noven displays 'Latitude in degrees', 'Longitude in degrees', etc. when the location/Y/value is selected.  For an event csv with an example row that looks like this:
`Array,Shot-ID(ids),Latitute (deg),Longitude (deg),Elev (m),Shot-Time (UTC)`
`2,2003,47.60093889,-122.3129306,638.7,2019:195:20:20:23.09028809`

the saved kef file then has rows that look like this example row:

`
        location/X/value_d = -122.3129306`
        `location/X/units_s = degrees`
        `location/Y/value_d = 47.60093889`
        `location/Y/units_s = degrees`
        `id_s = 2003`
        `location/Z/value_d = 638.7`
        `location/Z/units_s = m`
        `time/ascii_s = Sun Jul 14 20:20:23 2019`
        `time/epoch_l = 1563135623`
        `time/micro_seconds_i = 90287`
        `time/type_s = BOTH
`

I believe noven works the same way for location value and units for a receiver/station csv becoming a kef file.  Because noven automatically assigns them, I think the situation of missing units would be unusual.  I think we should either flag missing units for the event and array tables, or flag neither, just to be consistent.

This may raise the question of whether we want noven to automatically assign the units, even though lat & long in degrees and elevation (or depth) in meters is probably the standard we do want in PH5 archives.

Reply:
It doesn't seem like this warning would trigger if `location/X[Y,Z]/units_s = degrees` are assigned. These fields have a value of degrees. This check is looking for location/X[Y,Z]/units_s = '', which does seem like an error rather than a warnings. 

We should flag missing unit in both the event and array table as errors. I agree we need to be consistent with this. These are data issues that need to be fixed when the data is first generated. 

We may want to be able to change  lat & long units to meters, but I think this can be done manually and it seems like there are more pressing issues than updating the GUI.

Reply:
After reinspecting this issue it seems alright to leave location/X[Y,Z]/units_s = '' as a warnings because it occurs infrequently.